weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/DatasetDrawerContext.tsx

```
@@ -0,0 +1,475 @@
+    case ACTION_TYPES.SET_DATASET_OBJECT:
+      return {...state, datasetObject: action.payload};
+
+    case ACTION_TYPES.SET_SOURCE_SCHEMA:
+      return {...state, sourceSchema: action.payload};
+
+    case ACTION_TYPES.SET_TARGET_SCHEMA: {
+      // When the target schema is set (after dataset selection), suggest field mappings
+      const mappings = suggestFieldMappings(
+        state.sourceSchema,
+        action.payload,
+        state.selectedDataset?.objectId === state.selectedDataset?.objectId
```

Biome (1.9.4)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)